### PR TITLE
List:  Added localization for list selectAll checkbox state (T1248140)

### PR DESCRIPTION
**Pull request by nikkithelegendarypokemonster**

### CODE REVIEW

File: packages/devextreme/js/localization/messages/ar.json [hunk 1]
```diff
@@ -48,6 +48,9 @@
         "dxList-listAriaLabel-deletable": "Deletable items",
         "dxListEditDecorator-delete": "حذف",
         "dxListEditDecorator-more": "المزيد",
+        "dxList-halfChecked": "half checked",
```
Comment (ksercs) [inline packages/devextreme/js/localization/messages/ar.json]: List itself cannot be checked, only its selectAll checkbox. I think these variables should be like this:
```json
 "dxList-selectAll-indeterminate": "half checked",
 "dxList-selectAll-checked": "checked",
 "dxList-selectAll-notChecked": "not checked",
```
But before applying changes let's get approve by @vladaskorohodova

File: packages/devextreme/js/__internal/ui/list/m_list.edit.decorator.selection.ts [hunk 2]
```diff
@@ -185,8 +185,9 @@ registerDecorator(
 
       const indeterminate = value === undefined;
 
-      const checkedText = indeterminate ? 'half checked' : value ? 'checked' : 'not checked';
-      const label = `${messageLocalization.format('dxList-selectAll')}, ${checkedText}`;
+      const checkedText = indeterminate ? 'halfChecked' : value ? 'checked' : 'notChecked';
```
Comment (ksercs) [inline packages/devextreme/js/__internal/ui/list/m_list.edit.decorator.selection.ts]:
```suggestion
      const stateVariableName = indeterminate ? 'indeterminate' : value ? 'checked' : 'notChecked';
```

File: packages/devextreme/js/__internal/ui/list/m_list.edit.decorator.selection.ts [hunk 3]
```diff
@@ -185,8 +185,9 @@ registerDecorator(
...
+
+      const label = `${messageLocalization.format('dxList-selectAll')}, ${messageLocalization.format(`dxList-${checkedText}`)}`;
```
Comment (ksercs) [inline packages/devextreme/js/__internal/ui/list/m_list.edit.decorator.selection.ts]:
```suggestion
      const label = `${messageLocalization.format('dxList-selectAll')}, ${messageLocalization.format(`dxList-selectAll-${stateVariableName}`)}`;
```

File: packages/devextreme/js/localization/messages/ar.json (hunk 1 above)
Comment (vladaskorohodova) [reply packages/devextreme/js/localization/messages/ar.json]: Please redo as follows:
```
 "dxList-selectAll-indeterminate": "Half-checked",
 "dxList-selectAll-checked": "Checked",
 "dxList-selectAll-notChecked": "Not checked",
```

File: packages/devextreme/js/__internal/ui/list/m_list.edit.decorator.selection.ts (hunk 3 above)
Comment (nikkithelegendarypokemonster) [reply packages/devextreme/js/__internal/ui/list/m_list.edit.decorator.selection.ts]: [54d0fca](https://github.com/DevExpress/DevExtreme/pull/28070/commits/54d0fca21aee869cdc5fd4ccaf7d633ab9fc0873)

File: packages/devextreme/js/__internal/ui/list/m_list.edit.decorator.selection.ts (hunk 2 above)
Comment (nikkithelegendarypokemonster) [reply packages/devextreme/js/__internal/ui/list/m_list.edit.decorator.selection.ts]: [54d0fca](https://github.com/DevExpress/DevExtreme/pull/28070/commits/54d0fca21aee869cdc5fd4ccaf7d633ab9fc0873)

File: packages/devextreme/js/localization/messages/ar.json (hunk 1 above)
Comment (nikkithelegendarypokemonster) [reply packages/devextreme/js/localization/messages/ar.json]: [54d0fca](https://github.com/DevExpress/DevExtreme/pull/28070/commits/54d0fca21aee869cdc5fd4ccaf7d633ab9fc0873)

**Review by ksercs — APPROVED** [review]

**Review by vladaskorohodova — APPROVED** [review]

**Pull request merged**